### DISCUSSION TITLE
可以将sqlparse生成的ast单独出来？

### DISCUSSION THREAD

**Discussion by DLuPan** [apache/doris#6784] [Ideas]
使用doris一般企业都会有自己的一套DDL管理平台，通过DDL来实现对建表的管控，需要对DDL进行解析，当前doris fe的sqlparse集成太多的东西包含thrift和analysis等等还有一堆业务代码，导致生成AST很不干净，而且没法独立出来使用。
当前我的解决方案使用最新版本的sql_parser.cup和sql_scanner.flex实现的SQL解析，但是后续doris升级维护成本很高。doris可以将sqlparser单独作为服务，然后使用vistor模型进行analysis就好了。

**Comment by EmmyMiao87** [comment]
doris 是有 audit log 的，一般来说如果需要审计 sql 的话，可以利用 doris 的audit log。

**Reply by DLuPan** [reply]
数据治理这边需要统一建表，需要解析业务输入的sql的库、表、字段是否符合规范同时会自动添加敏感属性及执行等级，这些都是要在sql执行前进行处理。

**Reply by DLuPan** [reply]
https://github.com/DLuPan/DorisParser

这个是我现在提取出来的CREATE+ALTER部分的内容，希望官方能够提供解析用的jar包这样doris更新我们就不用同步去更新语法规则了
